tests/instruments/lecroy/test_lecroyT3DSO1204.py

```diff
@@ -0,0 +1,345 @@
+        instr.memory_size = 14e5
+        assert instr.memory_size == 7e3
+        instr.memory_size = 7e3
+
+
+def test_sample_size():
+    with expected_protocol(
+            LeCroyT3DSO1204,
+            [("CHDR OFF", None),
+             (b"SANU? C1", b"3.50E+06"),
+             (b"SANU? C1", b"3.50E+06"),
+             (b"SANU? C3", b"3.50E+06"),
+             (b"SANU? C3", b"3.50E+06"),
+             ]
+    ) as instr:
+        assert instr.acquisition_sample_size_c1 == 3.5e6
+        assert instr.acquisition_sample_size_c2 == 3.5e6
+        assert instr.acquisition_sample_size_c3 == 3.5e6
+        assert instr.acquisition_sample_size_c4 == 3.5e6
+
+
+def test_waveform_preamble():
+    with expected_protocol(
+            LeCroyT3DSO1204,
+            [("CHDR OFF", None),
+             (b"WFSU?", b"SP,1,NP,0,FP,0"),
+             (b"ACQW?", b"SAMPLING"),
+             (b"SARA?", b"1.00E+09"),
+             (b"SAST?", b"Trig'd"),
+             (b"MSIZ?", b"7M"),
+             (b"TDIV?", b"5.00E-04"),
+             (b"TRDL?", b"-0.00E+00"),
+             (b"SANU? C1", b"1.75E+06"),
+             (b"C1:VDIV?", b"5.00E-02"),
+             (b"C1:OFST?", b"-1.50E-01"),
+             ]
+    ) as instr:
+        assert instr.waveform_preamble == {
+            "sparsing": 1,
+            "requested_points": 0,
+            "transmitted_points": None,
+            "sampled_points": 1.75e6,
+            "first_point": 0,
+            "memory_size": 7e6,
+            "source": "C1",
+            "type": "normal",
+            "average": None,
+            "sampling_rate": 1e9,
+            "grid_number": 14,
+            "status": "triggered",
+            "xdiv": 5e-4,
+            "xoffset": 0,
+            "ydiv": 0.05,
+            "yoffset": -0.150
+        }
+
+
+def test_download_too_much_data():
+    with expected_protocol(
+            LeCroyT3DSO1204,
+            [("CHDR OFF", None),
+             (b'ACQW SAMPLING', None),
+             (b'SANU? C1', b'7.00E+06'),
+             ]
+    ) as instr:
+        with pytest.raises(ValueError):
+            instr.download_data(source="c1", requested_points=1e10)
+
+
+@pytest.mark.skip("Waiting for PR 660 to be merged so that the binary_values method is mocked")
+def test_download_data_until_digitize():
+    with expected_protocol(
+            LeCroyT3DSO1204,
+            [("CHDR OFF", None),
+             (b'ACQW SAMPLING', None),
+             (b'SANU? C1', b'7.00E+06'),
+             (b"WFSU SP,1", None),
+             (b"WFSU NP,1", None),
+             (b"WFSU FP,0", None),
+             (b"WFSU?", b"SP,1,NP,1,FP,0"),
+             (b"ACQW?", b"SAMPLING"),
+             (b"SARA?", b"1.00E+09"),
+             (b"SAST?", b"Stop"),
+             (b"TDIV?", b"5.00E-04"),
+             (b"TRDL?", b"-0.00E+00"),
+             (b"C1:VDIV?", b"5.00E-02"),
+             (b"C1:OFST?", b"-1.50E-01"),
+             (b"C1:WF? DAT2", b"DAT2,#90000001" + (0x01).to_bytes(2, byteorder='big') + b"\n\n"),
```

Comment:
Do you want to show, how you got there explicitly? If not, you can just convert it to bytes (similarly two lines below):
```suggestion
             (b"C1:WF? DAT2", b"DAT2,#90000001" + b"\x00\x01" + b"\n\n"),
```

Reply:
Thank you for the suggestion. Applied in commit cfce4a3815759a85811cc1231bbc7d42fa124254